src/librbd/managed_lock/BreakRequest.h

```diff
@@ -55,9 +61,13 @@ class BreakRequest {
    * @endvertbatim
    */
 
-  ImageCtxT &m_image_ctx;
+  librados::IoCtx &m_ioctx;
+  CephContext *m_cct;
+  ContextWQ *m_work_queue;
+  std::string m_oid;
   const Locker &m_locker;
```

Comment:
I'd be fine if you changed it to support your use-case. W/ the original use-case, there was no race since the caller wouldn't be touching the memory while the call was in-progress.